src/main/java/com/nitnelave/CreeperHeal/block/CreeperBed.java

```diff
@@ -1,90 +1,62 @@
 package com.nitnelave.CreeperHeal.block;
 
-import org.bukkit.Material;
-import org.bukkit.block.Block;
-import org.bukkit.block.BlockFace;
 import org.bukkit.block.BlockState;
-import org.bukkit.material.Bed;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.block.data.type.Bed;
 
-import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 /**
- * Bed implementation of CreeperBlock.
- * 
- * @author nitnelave
- * 
+ * Bed implementation of CreeperMultiblock.
+ *
+ * @author Jikoo
  */
-class CreeperBed extends CreeperBlock
+class CreeperBed extends CreeperMultiblock
 {
 
-    /*
-     * The direction the bed is facing.
-     */
-    private final BlockFace orientation;
-
-    /*
-     * Constructor.
-     */
     CreeperBed(BlockState blockState)
     {
-        Bed bedData = castData(blockState, Bed.class);
-        orientation = bedData.getFacing();
-        Block block = blockState.getBlock();
-        if (!bedData.isHeadOfBed())
-            block = block.getRelative(orientation);
-        this.blockState = block.getState();
-    }
-
-    /**
-     * The blockstate is always the head of the bed, this gets the foot.
-     *
-     * @return the foot of the bed.
-     */
-    public Block getFoot()
-    {
-        return getBlock().getRelative(orientation.getOppositeFace());
+        super(blockState);
+
+        BlockData blockData = blockState.getBlockData();
+        if (!(blockData instanceof Bed))
+            throw new IllegalArgumentException("Invalid BlockData: " + blockData.getClass().getName());
```

Comment:
Are you sure that's not supposed to happen? I.e. there are no "normal" cases where this could throw an exception? Users don't like exceptions.

Reply:
Since we [check](https://github.com/nitnelave/CreeperHeal/pull/60/commits/42787f31159949b2eaec5bce05e59abe03c3e079#diff-bbe1d79e8082a86849d6b8c5c3d31b69R92) if it's a Bed prior to creation, yes. Someone's gotta be doing something exceptionally weird.